lib/SILOptimizer/Transforms/FunctionSignatureOpts.cpp

```diff
@@ -681,13 +681,21 @@ void FunctionSignatureTransform::createFunctionSignatureOptimizedFunction() {
     Arg->setOwnershipKind(Ownershipkind);
   }
 
-  // Create the thunk body !
+  // Create the thunk body!
   F->setThunk(IsThunk);
   // The thunk now carries the information on how the signature is
   // optimized. If we inline the thunk, we will get the benefit of calling
   // the signature optimized function without additional setup on the
   // caller side.
   F->setInlineStrategy(AlwaysInline);
+  // If the new callee is @_semantics("stdlib_binary_only"), it is fine to call
+  // it, because it will be exposed as a public symbol in the object file.
+  if (F->hasSemanticsAttr("stdlib_binary_only")) {
```

Comment:
This should instead be encoded as the function having IsNotSerialized set. Why do we need to specially handle it here?